### PR TITLE
[CH-4014] Upgrade ttlcache to use generics

### PR DESCRIPTION
**Pull request by chrishaines**
This upgrades the workers-pool package to use v3 of `ttlcache` which supports generics. It also updates the package to use Go Modules.

### CODE REVIEW

**Review by christorbraze — APPROVED** [review]
I guess all of these changes (using `Value`, `Delete`, `Set`) are due to the ttlcache library being ugpraded? Did we also modernize the build, accounting for `makefile`, `go.mod`, etc?

**Comment by chrishaines** [comment]
> I guess all of these changes (using `Value`, `Delete`, `Set`) are due to the ttlcache library being ugpraded? Did we also modernize the build, accounting for `makefile`, `go.mod`, etc?

Yup on both counts

File: makefile [hunk 1]
```diff
@@ -1,23 +1,31 @@
...
+export PATH := $(GOBIN):$(PATH)
+
 .PHONY: setup
 setup: ## Install all the build and lint dependencies
 	go get -u github.com/golangci/golangci-lint/cmd/golangci-lint
 	go get -u github.com/golang/dep/cmd/dep
 	go get -u golang.org/x/tools/cmd/cover
 	go get -u golang.org/x/tools/cmd/goimports
 	go get -u github.com/AlekSi/gocoverutil
-	dep ensure
```
Comment (froodian) [inline makefile:12]: can this have a 

```
GOPRIVATE=$(GOPRIVATE) go mod tidy
GOPRIVATE=$(GOPRIVATE) go mod download
```

this should a) shrink the go.mod and b) ensure everything in the go.mod is installed

File: makefile [hunk 2]
```diff
@@ -26,7 +34,19 @@ fmt: ## gofmt and goimports all go files
...
+
+.PHONY: ci
+ci: test lint ## Run all the tests and code checks
+
+.PHONY: install
+install: ## Install to $GOPATH/src
+	go install ./...
+
+.PHONY: package
+package: ## Package the executable to PACKAGE_DIR
```
Comment (froodian) [inline makefile]: is this really necessary, since this doesn't get installed as its own executable, it just gets used a library?

**Review by froodian — APPROVED** [review]

**Review by MiltonZu — APPROVED** [review]
Looks good.  Definitely less verbose and easier to read.

**Pull request merged**